app/models/borrow_history.rb

```diff
@@ -0,0 +1,52 @@
+class BorrowHistory < ApplicationRecord
+  include Api::V1::BorrowHistoriesHelper
+  extend Mobility
+
+  belongs_to :user
+  belongs_to :book
+  has_one :approved_by, class_name: :AdminUser , foreign_key: "admin_user_id"
```

Comment:
try to combine associations in a part , validations in a part and scopes , enums together not all over the place